Marriage Fix - Lynda

EeFo17 opened the pull request:
Marriage is not working.

# Description

Now you need a wedding ring and a wedding dress, Lynda add wedding dresses and rings with a special description.

## Behaviour
### **Actual**

At the moment, the only problem is removing the outfit in divorce. Only one partner loses outfit.

### **Expected**

After divorce players should loses wedding outfits.

## Fixes

\# (issue)

## Type of change

Please delete options that are not relevant.

  - [ x ] Bug fix (non-breaking change which fixes an issue)
  - [ x ] New feature (non-breaking change which adds functionality)

**Test Configuration**:

  - Server Version: 1.0.5
  - Client: 12.72
  - Operating System: Ubuntu 

## Checklist

  - [ ] My code follows the style guidelines of this project
  - [ ] I have performed a self-review of my own code
  - [ ] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [ ] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

EPuncker commented:
da rebase ou cria um branch novo

gpedro commented on data/movements/scripts/quests/demon oak/entrance.lua:
```diff
@@ -22,10 +22,12 @@ function onStepIn(creature, item, position, fromPosition)
 	-- CHECAR SE TA TUDO CERTO Vou usar mensagens pra descobrir qual � o conflito
 	if (#Game.getSpectators(DEMON_OAK_POSITION, false, true, 9, 9, 6, 6) == 0) then
 			if (player:getItemCount(10305) == 0) then
+				if player:getStorageValue(Storage.DemonOak.Progress) < 1 then
```
can you merge this conditions?

EeFo17 committed:
Marriage Fix - Lynda 

Marriage is not working.
Now you need a wedding ring and a wedding dress, Lynda add wedding dresses and rings with a special description.

At the moment, the only problem is removing the outfit in divorce. Only one partner loses outfit.

EeFo17 committed:
Update lynda.lua

EeFo17 committed:
Update lynda.lua

EeFo17 committed:
Update global.lua

EeFo17 committed:
Update player.lua

worthdavi commented on data/events/scripts/player.lua:
```diff
@@ -199,6 +199,11 @@ function Player:onLook(thing, position, distance)
 	end
+		if LOOK_MARRIAGE_DESCR and thing:isCreature() then
+        if thing:isPlayer() then
+            description = description .. self:getMarriageDescription(thing)
```
self:getMarriageDescription() is calling getPlayerSpouse(), getPlayerMarriageStatus() and getPlayerNameById()
3 db queries in a onLook function

I think this function would be way easier to make in player.cpp/h

doenteverde replied:
How cool are you going to post or just said?  
kk

worthdavi replied:
Sorry, I don't get you. But if you meant if am I going to post some code, no. He asked for a hint/advice to improve the code, and I gave mine.

dudantas commented on data/npc/scripts/lynda.lua:
```diff
@@ -126,44 +126,96 @@ local function confirmWedding(cid, message, keywords, parameters, node)
+end
+
+local function tryEngage(cid, message, keywords, parameters, node)
+    if(not npcHandler:isFocused(cid)) then
```
These functions no longer exist, the lib has been modified

dudantas commented on data/npc/scripts/lynda.lua:
```diff
@@ -126,44 +126,96 @@ local function confirmWedding(cid, message, keywords, parameters, node)
+        msg = msg .. ' What brings you here? Wanting to {marry} someone?'
+    end
+    npcHandler:say(msg,cid)
+    npcHandler:addFocus(cid)
```
Same

dudantas reviewed (changes requested):
You need to edit the npc to use npcs Lua new lib in revscript
